src/framework/parsers/draco-decoder.js

```diff
@@ -0,0 +1,168 @@
+import { WasmModule } from "../../core/wasm-module";
+import { ABSOLUTE_URL } from "../asset/constants";
+import { DracoWorker } from "./draco-worker";
+
+class JobQueue {
+    constructor(workers) {
+        this.workers = [workers, [], []];
```

Comment:
what are we storing here? Could those be separately stored arrays for clarity, and avoiding things like `this.workers[2]`?

Reply:
or maybe some of those should be Sets, considering you do indexOf on them

Reply:
`workers` stores 3 arrays of workers depending on how many jobs they're actively working on.

`workers[0]` contains workers with no jobs.
`workers[1]` contains workers with 1 job queued.
`workers[2]` contains workers with 2 jobs queued.

We want to keep worker order in each array so we push to the back and splice from the front of the array.

Reply:
ah I didn't guess that, I assumed [0] is free, [1] is busy, [2] is done or something like that.
We need documentation.